### ISSUE TITLE
ShapeSource.onPress event not preserving elevation in coordinates array

### ISSUE THREAD

**Issue by mzu** [react-native-mapbox-gl/maps#1092]
**Describe the bug**
According to the [GeoJSON spec](https://geojson.org/geojson-spec.html#positions) adding additional values to the coordinates apart from long,lat is valid . In my case I use a service that adds the elevation as a 3rd element. 

The GeoJSON might look like this:

```js
const featureCollection = {
  type: 'FeatureCollection',
  features: [
    {
      type: 'Feature',
      id: '9d10456e-bdda-4aa9-9269-04c1667d4552',
      properties: {
        name: 'Test Icon 1',
        icon: 'example',
      },
      geometry: {
        type: 'Point',
        coordinates: [-117.20611157485, 52.180961084201, 873.1],
      },
    },
  ],
};
```
I noticed that the onPress() event of ``MapboxGL.ShapeSource`` replaces or strips the coordinates of the feature.

For example pressing the ShapeSource icon for

```js
          <MapboxGL.ShapeSource
            id="exampleShapeSource"
            shape={featureCollection}
            onPress={(e) => this.onPressIcon(e.features[0])}>
            <MapboxGL.SymbolLayer id="exampleIconName" style={styles.icon} />
          </MapboxGL.ShapeSource>
```

with 

```js
  onPressIcon(feature) {
    Alert.alert(
      feature.properties.name,
      JSON.stringify(feature.geometry.coordinates),
    );
  }
```

will result in only showing the 2 first elements of the coordinates array in the alert message and the elevation data is lost.

Is there a way to avoid this behaviour? Is this a bug? As I understand it is very common to keep the elevation with the coordinates. The workaround would be to put the elevation into properties.

**To Reproduce**
I created a test branch inside my fork for the example app and changed [ShapeSourceIcon.js](https://github.com/mzu/maps/blob/test-branch/example/src/examples/ShapeSourceIcon.js) to demonstrate the issue.

**Expected behavior**
The elevation inside the coordinates array should be preserved.

**Versions (please complete the following information):**
 - Platform tested : iOS
 - Device:  iPhone11
 - Emulator/ Simulator: no
 - OS: 14.0.1
 - react-native-mapbox-gl Version: tested with 8.1.0-rc.8 and 8.1.0-rc.4
 - React Native Version 0.63.3

**Comment by ferdicus** [comment]
Disclaimer: All digging done in Java (but I checked your bug also occurs there)
hey there, I did some digging.... 

so the `feature` is correctly passed to the native RCTMGLShapeSource component which can be observed when you check the code here via debugger: https://github.com/react-native-mapbox-gl/maps/blob/693852e5f01f5cbc05b97f63a5dce6ebea6903e0/android/rctmgl/src/main/java/com/mapbox/rctmgl/components/styles/sources/RCTMGLShapeSource.java#L92-L98
[Line 93](https://github.com/react-native-mapbox-gl/maps/blob/master/android/rctmgl/src/main/java/com/mapbox/rctmgl/components/styles/sources/RCTMGLShapeSource.java#L93)


Then the culprit could be one of two methods:
* either [`.setGeoJson(mShape)`](https://github.com/react-native-mapbox-gl/maps/blob/master/android/rctmgl/src/main/java/com/mapbox/rctmgl/components/styles/sources/RCTMGLShapeSource.java#L96) is somehow not setting things up correctly
* or the method [`.queryRenderedFeatures`](https://github.com/react-native-mapbox-gl/maps/blob/master/android/rctmgl/src/main/java/com/mapbox/rctmgl/components/mapview/RCTMGLMapView.java#L602) used in [RCTMGLMapView.java](https://github.com/react-native-mapbox-gl/maps/blob/master/android/rctmgl/src/main/java/com/mapbox/rctmgl/components/mapview/RCTMGLMapView.java) isn't correctly returning the feature based on our shape in ShapeSource

Notice, this is based on my limited knowledge of Java and this repo, however that's what I got from some debugging. 
Maybe @mfazekas, can shed some more light on this and let me know if I'm completely off base here. 

However, if all the things I wrote hold true - this is most like an issue upstream and not with us.

Cheers

**Comment by mzu** [comment]
OK, thanks. If it is upstream will you normally let them know? This one is not critical since there is an easy workaround. Just thought it makes sense to document this since it's not working according to what some people might expect when working with GeoJSON.

**Comment by ferdicus** [comment]
I would leave it to the discretion of the user who opened the ticket if it isn't critical.

**Comment by mzu** [comment]
Happy to pass this upstream if indeed a `com.mapbox.mapboxsdk` issue. Will wait for confirmation from @mfazekas first.

**Comment by mfazekas** [comment]
I agree this is most likely an unimplemented feature from upstream.

**Comment by mzu** [comment]
Just one more thing I noticed in relation to this ticket. In the original description of this issue, I described the following:

> I noticed that the onPress() event of MapboxGL.ShapeSource replaces or strips the coordinates of the feature.

It is actually more than that. The whole `geometry` part of the feature that gets passed in the event gets overwritten. The `coordinates` from the event feature do not match the original feature from the source anymore. I now think that this is a bit more critical than I first thought.

In my case this means if I click a `Point` on a `Line` where the point's coordinates matches the underlying vertex of the line, the actual feature from the `onPress()` event will not match the vertex from the line anymore.

Wouldn't it be better to at least keep the original `geometry` inside the feature (maybe renamed as `geometry_source`) together with the geometry from the event? This would then also handle the issue with losing the elevation info. 

I haven't raised this as an issue with the upstream repository yet since I do not know enough about this side to point out the places they need to look at.
